frodo.js

```diff
@@ -8,85 +8,94 @@
  *
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.